docs/proposals/assets/multiple-dns-provider-support/google/google.md

```diff
@@ -0,0 +1,107 @@
+## Google
+
+https://console.cloud.google.com/net-services/dns/zones
+
+Google supports Weighted(Weighted Round Robin) and Geolocation routing policies https://cloud.google.com/dns/docs/zones/manage-routing-policies. Both of these can be configured directly on records in Google Cloud DNS and no secondary Traffic Management resource is required.
+
+Notes:
+
+* Record sets are modelled as a single endpoint with routing policy embedded. This is a different approach to Route53 where each individual A/CNAME would have its own record entry. 
+* Weight must be an integer between 0 - 10000
+* There are no continent options for region, only finer grained regions such as us-east1, europe-west -1 etc...
+* There appears to be no way to set a default region, google just routes requests to the nearest supported region.
+* The current approach used in AWS Route53 for geo routing will work in the same way on Google DNS. A single CNAME record with geo routing policy specifying multiple geo specific A record entries as targets.
+* Geo and weighted routing can be combined, as with AWS Route53, allowing traffic with a region to be routed using weightings.
+* The modelling of the records is slightly different from AWS, but the current DNSRecord spec could still work. The google implemention of AddRecords will have to process the list of endpoints in order to group related endpoints in order to build up the required API request.
+In this case there would not be a 1:1 mapping between an endpoint in a DNSREcord and the dns provider, but the DNSRecord contents would be kept consistent accross all providers and compatability with external-dns would be maintained.
```

Comment:
> Consistency would be more important than external-dns compatibility do you think? Are there changes to the DNSRecord resource you'd recommend that would make consistency easier?

Yes , external-dns compatibility isn't of high importance and shouldn't stop us modifying the spec if it make sense to us. It was just an observation, as being able to commit back to external-dns was something i had hoped we would be able to do, and not changing the Endpoints spec would likely make that easier and more likely to be accepted. 

We could make changes to the spec to allow a more provider specific set of endpoints to be defined, but whether they will ultimately be better or not I'm not sure. It might actually be better to keep it generic and have the implementation modify the endpoint list as required to make the necessary requests, and document what will be created in each provider. This especially the case with Azure that requires additional resources to be created.

I think ultimately we can decide this as we implement it and we get a better idea of what makes sense.